src/core/dbus-execute.c

```diff
@@ -3962,6 +3981,46 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
+        } else if (STR_IN_SET(name, "StateDirectoryQuota", "CacheDirectoryQuota", "LogsDirectoryQuota")) {
+                uint64_t quota_absolute;
+                uint32_t quota_scale;
+                bool quota_set;
+
+                r = sd_bus_message_enter_container(message, 'r', "tub");
+                if (r < 0)
+                        return r;
+
+                r = sd_bus_message_read(message, "t", &quota_absolute);
+                if (r < 0)
+                        return r;
+
+                r = sd_bus_message_read(message, "u", &quota_scale);
+                if (r < 0)
+                        return r;
+
+                r = sd_bus_message_read(message, "b", &quota_set);
+                if (r < 0)
+                        return r;
+
+                r = sd_bus_message_exit_container(message);
+                if (r < 0)
+                        return r;
+
+                ExecDirectoryType dt;
+                dt = exec_directory_type_from_string(name);
+                if (dt < 0)
+                        return -EINVAL;
+
+                if (!UNIT_WRITE_FLAGS_NOOP(flags)) {
+                        c->directories[dt].exec_quota.quota_absolute = quota_absolute;
+                        c->directories[dt].exec_quota.quota_scale = quota_scale;
+                        c->directories[dt].exec_quota.quota_set = quota_set;
+
+                        unit_write_settingf(u, flags, name, "%s=%lu %u %d", name, quota_absolute, quota_scale, quota_set);
```

Comment:
this is not right. you need to write out a unit file setting, i.e. something config_parse_exec_quota() can actually parse